joern-cli/frontends/swiftsrc2cpg/src/main/scala/io/joern/swiftsrc2cpg/astcreation/AstForDeclSyntaxCreator.scala

```diff
@@ -34,7 +36,58 @@ trait AstForDeclSyntaxCreator(implicit withSchemaValidation: ValidationMode) {
     astForTypeDeclSyntax(node)
   }
 
-  private def astForAssociatedTypeDeclSyntax(node: AssociatedTypeDeclSyntax): Ast = notHandledYet(node)
+  private def astForAssociatedTypeDeclSyntax(node: AssociatedTypeDeclSyntax): Ast = {
+    // TODO:
+    // - handle genericWhereClause
+    val attributes = astForDeclAttributes(node)
+    val modifiers  = modifiersForDecl(node)
+    val aliasName  = node.initializer.map(i => code(i.value))
+
+    val name                     = typeNameForDeclSyntax(node)
+    val (typeName, typeFullName) = calcTypeNameAndFullName(name)
+    registerType(typeFullName)
+
+    val existingTypeDecl = global.seenTypeDecls.keys().asScala.find(_.name == typeName)
+    if (existingTypeDecl.isEmpty) {
+
+      val astParentType     = methodAstParentStack.head.label
+      val astParentFullName = methodAstParentStack.head.properties("FULL_NAME").toString
```

Comment:
`import io.shiftleft.codepropertygraph.generated.PropertyNames`
```suggestion
      val astParentFullName = methodAstParentStack.head.properties(PropertyNames.FULL_NAME).toString
```